soroban-env-host/observations/test__stellar_asset_contract__verify_nested_try_call_rollback.json

```diff
@@ -1,480 +1,480 @@
 {
   "   0 begin": "cpu:0, mem:0, prngs:-/d8a09d57, objs:-/-, vmem:-, evt:-, store:-/-",
-  "   1 call upload_wasm(Bytes(obj#1))": "cpu:3344, mem:2377, objs:-/1@32cc0b00",
-  "   2 ret upload_wasm -> Ok(Bytes(obj#3))": "cpu:2333606, mem:1341893, objs:-/2@c00fda96, store:-/1@1e90209",
-  "   3 call create_contract(Address(obj#5), Bytes(obj#3), Bytes(obj#7))": "cpu:2337536, mem:1342069, objs:-/4@1a5f766",
-  "   4 call obj_cmp(Address(obj#9), Address(obj#5))": "cpu:2342197, mem:1342247, objs:-/5@bf99e74c",
-  "   5 ret obj_cmp -> Ok(0)": "cpu:2342883",
-  "   6 call get_ledger_network_id()": "cpu:2343141",
-  "   7 ret get_ledger_network_id -> Ok(Bytes(obj#11))": "cpu:2345930, mem:1342359, objs:-/6@acef0e26",
-  "   8 ret create_contract -> Ok(Address(obj#13))": "cpu:2383687, mem:1343905, objs:-/7@f36723d7, store:-/2@6ee999fd",
-  "   9 call upload_wasm(Bytes(obj#15))": "cpu:2387072, mem:1346585, objs:-/8@a19bb4fb",
-  "  10 ret upload_wasm -> Ok(Bytes(obj#17))": "cpu:4896088, mem:2698801, objs:-/9@6c1f0b74, store:-/3@7eec256f",
-  "  11 call create_contract(Address(obj#19), Bytes(obj#17), Bytes(obj#21))": "cpu:4900018, mem:2698977, objs:-/11@3280bab9",
-  "  12 call obj_cmp(Address(obj#23), Address(obj#19))": "cpu:4904679, mem:2699155, objs:-/12@9872f492",
-  "  13 ret obj_cmp -> Ok(0)": "cpu:4905365",
-  "  14 call get_ledger_network_id()": "cpu:4905623",
-  "  15 ret get_ledger_network_id -> Ok(Bytes(obj#25))": "cpu:4908412, mem:2699267, objs:-/13@4635c7d8",
-  "  16 ret create_contract -> Ok(Address(obj#27))": "cpu:4951939, mem:2701185, objs:-/14@9056dd8b, store:-/4@bdc76bac",
-  "  17 call create_asset_contract(Bytes(obj#29))": "cpu:4962109, mem:2701942, objs:-/15@337e5509, store:-/5@ea816a69",
-  "  18 call get_ledger_network_id()": "cpu:4963480, mem:2702002",
-  "  19 ret get_ledger_network_id -> Ok(Bytes(obj#31))": "cpu:4966269, mem:2702114, objs:-/16@12b2a48f",
-  "  20 call symbol_len(Symbol(obj#33))": "cpu:5015595, mem:2704478, objs:-/18@d2537e93, store:-/6@b0279aa9",
-  "  21 ret symbol_len -> Ok(U32(10))": "cpu:5015703",
-  "  22 push SAC:baaaf5fd:sym#33(Bytes(obj#35))": "cpu:5042686, mem:2705894, objs:-/19@2c66bacd",
-  "  23 call vec_new()": "cpu:5048782",
-  "  24 ret vec_new -> Ok(Vec(obj#39))": "cpu:5049923, mem:2705958, objs:-/20@c0a3859",
+  "   1 call upload_wasm(Bytes(obj#1))": "cpu:3231, mem:1520, objs:-/1@c24f5200",
```

Comment:
Do we understand why has the object hash change?

Reply:
the uploaded wasm changed (because they were regenerated with much more aggressive LTO). You can see the divergence happens right at upload_wasm.